src/ServiceManagement/Compute/Commands.ServiceManagement/Extensions/Common/ExtensionRole.cs

```diff
@@ -63,12 +78,16 @@ public override string ToString()
 
         public string GetExtensionId(string extensionName, string slot, int index)
         {
-            var normalizedExtName = extensionName.Replace(".", string.Empty);
-            normalizedExtName = normalizedExtName.Replace(" ", string.Empty);
-            normalizedExtName = normalizedExtName.Replace("_", string.Empty);
+            var normalizedExtName = RemoveDisallowedCharacters(extensionName);
 
             var suffix = new StringBuilder();
             suffix.AppendFormat(ExtensionIdSuffixTemplate, normalizedExtName, slot, index);
+            if (suffix.Length > MaxSuffixLength)
+            {
+                int lenDiff = suffix.Length - MaxSuffixLength;
+                int startIndex = 1; // Suffix starts with '-'
+                suffix.Remove(startIndex + normalizedExtName.Length - lenDiff, lenDiff);
```

Comment:
Yes, it's actually covered, :-)... -- The test case is to validate that a very long extension name will be truncated in the suffix, while the other parts, i.e. slot & number are preserved.